components/builder-api/src/depot/server.rs

```diff
@@ -2020,6 +2168,16 @@ fn demote_package(req: &mut Request) -> IronResult<Response> {
         None => return Ok(Response::with(status::BadRequest)),
     };
 
+    // This stinks but is necessary since memcached Client returns mutable reference
+    // and iron chain.link returns everything wrapped in Arc
+    // The tradeoff here is that we have to initialize a new client on every request until
+    // we can switch to a framework that's not Iron
+    let depot = req.get::<persistent::Read<Config>>().unwrap();
+    let memcache_host_strings: Vec<String> =
+        depot.memcached.iter().map(|h| format!("{}", h)).collect();
+    let memcache_hosts: Vec<&str> = memcache_host_strings.iter().map(AsRef::as_ref).collect();
+    let mut client = memcache::Client::new(memcache_hosts).unwrap();
```

Comment:
That helper function is pretty much worth its weight in gold by this point in the file.